### ISSUE TITLE
lncli listpayments --include_incomplete incurs into a "cannot marshall failure *lnwire.InvalidOnionPayload" error

### ISSUE THREAD

**Issue by villastrangiato** [lightningnetwork/lnd#5173]
### Background

I've been trying twice to make some loop out without succeding. 
The result in both cases was leaving my node with some unsettled HTLCs, and in that situation I incurred in the error.

### Your environment

* version of `lnd`
lncli version 0.12.1-beta commit=v0.12.1-beta

* which operating system (`uname -a` on *Nix)
Linux XXXXX 5.11.11-200.fc33.x86_64 #1 SMP Tue Mar 30 16:53:32 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

* version of `btcd`, `bitcoind`, or other backend
Bitcoin Core version v0.21.0 (release build)

* any other relevant environment details
node running behind tor: Tor 0.4.5.7 running on Linux

### Steps to reproduce

Trying to loop out and wait to have some unsettled HTLCs.
issuing the command:

```
lncli listpayments  --include_incomplete
```

### Expected behaviour

getting a list of payments

### Actual behaviour

```
[lncli] rpc error: code = Unknown desc = cannot marshall failure *lnwire.InvalidOnionPayload
```

**Comment by Roasbeef** [comment]
That's odd, we have a case present to ensure we can encode the error type to the proto version of it: https://github.com/lightningnetwork/lnd/blob/master/lnrpc/routerrpc/router_backend.go#L1116

**Comment by Roasbeef** [comment]
Ah this was fixed here: https://github.com/lightningnetwork/lnd/pull/5074

But it isn't hasn't been packaged within a release yet. If you're able to, you can apply that commit/PR locally and then you'll be able to read out the data. Otherwise you can upgrade directly to master, but I'd recommend just applying that patch instead if you're able to.

**Comment by villastrangiato** [comment]
> Ah this was fixed here: #5074
> 
> But it isn't hasn't been packaged within a release yet. If you're able to, you can apply that commit/PR locally and then you'll be able to read out the data. Otherwise you can upgrade directly to master, but I'd recommend just applying that patch instead if you're able to.

great: applied the patch and it is working! thank you very much!
